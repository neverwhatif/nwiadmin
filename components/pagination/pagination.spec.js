import React from 'react';
import { shallow } from 'enzyme';

import Pagination from './index';

const mockLocation = {
    pathname: '/lorem',
    search: '',
};

const data = {
    from: 1,
    to: 10,
    last_page: 5,
    current_page: 1,
    total: 54,
};

const nonPaginationData = {
    to: 10,
    last_page: 1,
    current_page: 1,
    total: 54,
};

describe('Pagination', () => {
    const component = shallow(<Pagination data={data} location={mockLocation} />);
    const componentNonPagination = shallow(<Pagination data={nonPaginationData} location={mockLocation} />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
        expect(componentNonPagination).toMatchSnapshot();
    });
    it('should not render', () => {
        expect(componentNonPagination.type()).toBeNull();
    });
});
