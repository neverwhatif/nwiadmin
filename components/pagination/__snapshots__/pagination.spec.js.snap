// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    data={
      Object {
        "current_page": 1,
        "from": 1,
        "last_page": 5,
        "to": 10,
        "total": 54,
      }
    }
    location={
      Object {
        "pathname": "/lorem",
        "search": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PaginationSummary
          from={1}
          to={10}
          total={54}
        />,
        <ul
          className="list"
        >
          <li
            className="item"
          >
            <PaginationItem
              isCurrent={true}
              label={1}
              to="/lorem?page=1"
              type="link"
            />
          </li>
          <li
            className="item"
          >
            <PaginationItem
              isCurrent={false}
              label={2}
              to="/lorem?page=2"
              type="link"
            />
          </li>
          <li
            className="item"
          >
            <PaginationItem
              isCurrent={false}
              label={null}
              to={null}
              type="ellipsis"
            />
          </li>
          <li
            className="item"
          >
            <PaginationItem
              isCurrent={false}
              label={5}
              to="/lorem?page=5"
              type="link"
            />
          </li>
        </ul>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "from": 1,
          "to": 10,
          "total": 54,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="item"
            >
              <PaginationItem
                isCurrent={true}
                label={1}
                to="/lorem?page=1"
                type="link"
              />
            </li>,
            <li
              className="item"
            >
              <PaginationItem
                isCurrent={false}
                label={2}
                to="/lorem?page=2"
                type="link"
              />
            </li>,
            <li
              className="item"
            >
              <PaginationItem
                isCurrent={false}
                label={null}
                to={null}
                type="ellipsis"
              />
            </li>,
            <li
              className="item"
            >
              <PaginationItem
                isCurrent={false}
                label={5}
                to="/lorem?page=5"
                type="link"
              />
            </li>,
          ],
          "className": "list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <PaginationItem
                isCurrent={true}
                label={1}
                to="/lorem?page=1"
                type="link"
              />,
              "className": "item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "isCurrent": true,
                "label": 1,
                "to": "/lorem?page=1",
                "type": "link",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": <PaginationItem
                isCurrent={false}
                label={2}
                to="/lorem?page=2"
                type="link"
              />,
              "className": "item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "isCurrent": false,
                "label": 2,
                "to": "/lorem?page=2",
                "type": "link",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "ellipsis2",
            "nodeType": "host",
            "props": Object {
              "children": <PaginationItem
                isCurrent={false}
                label={null}
                to={null}
                type="ellipsis"
              />,
              "className": "item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "isCurrent": false,
                "label": null,
                "to": null,
                "type": "ellipsis",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "host",
            "props": Object {
              "children": <PaginationItem
                isCurrent={false}
                label={5}
                to="/lorem?page=5"
                type="link"
              />,
              "className": "item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "isCurrent": false,
                "label": 5,
                "to": "/lorem?page=5",
                "type": "link",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PaginationSummary
            from={1}
            to={10}
            total={54}
          />,
          <ul
            className="list"
          >
            <li
              className="item"
            >
              <PaginationItem
                isCurrent={true}
                label={1}
                to="/lorem?page=1"
                type="link"
              />
            </li>
            <li
              className="item"
            >
              <PaginationItem
                isCurrent={false}
                label={2}
                to="/lorem?page=2"
                type="link"
              />
            </li>
            <li
              className="item"
            >
              <PaginationItem
                isCurrent={false}
                label={null}
                to={null}
                type="ellipsis"
              />
            </li>
            <li
              className="item"
            >
              <PaginationItem
                isCurrent={false}
                label={5}
                to="/lorem?page=5"
                type="link"
              />
            </li>
          </ul>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "from": 1,
            "to": 10,
            "total": 54,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="item"
              >
                <PaginationItem
                  isCurrent={true}
                  label={1}
                  to="/lorem?page=1"
                  type="link"
                />
              </li>,
              <li
                className="item"
              >
                <PaginationItem
                  isCurrent={false}
                  label={2}
                  to="/lorem?page=2"
                  type="link"
                />
              </li>,
              <li
                className="item"
              >
                <PaginationItem
                  isCurrent={false}
                  label={null}
                  to={null}
                  type="ellipsis"
                />
              </li>,
              <li
                className="item"
              >
                <PaginationItem
                  isCurrent={false}
                  label={5}
                  to="/lorem?page=5"
                  type="link"
                />
              </li>,
            ],
            "className": "list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": <PaginationItem
                  isCurrent={true}
                  label={1}
                  to="/lorem?page=1"
                  type="link"
                />,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "isCurrent": true,
                  "label": 1,
                  "to": "/lorem?page=1",
                  "type": "link",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": <PaginationItem
                  isCurrent={false}
                  label={2}
                  to="/lorem?page=2"
                  type="link"
                />,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "isCurrent": false,
                  "label": 2,
                  "to": "/lorem?page=2",
                  "type": "link",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "ellipsis2",
              "nodeType": "host",
              "props": Object {
                "children": <PaginationItem
                  isCurrent={false}
                  label={null}
                  to={null}
                  type="ellipsis"
                />,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "isCurrent": false,
                  "label": null,
                  "to": null,
                  "type": "ellipsis",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "5",
              "nodeType": "host",
              "props": Object {
                "children": <PaginationItem
                  isCurrent={false}
                  label={5}
                  to="/lorem?page=5"
                  type="link"
                />,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "isCurrent": false,
                  "label": 5,
                  "to": "/lorem?page=5",
                  "type": "link",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Pagination should render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    data={
      Object {
        "current_page": 1,
        "last_page": 1,
        "to": 10,
        "total": 54,
      }
    }
    location={
      Object {
        "pathname": "/lorem",
        "search": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
